plasmapy/particles/_factory.py

```diff
@@ -9,15 +9,106 @@
 import astropy.units as u
 import contextlib
 
-from astropy.units.physical import electrical_charge, mass
-from numbers import Integral
-from typing import Union
+from astropy.constants import m_e
+from numbers import Integral, Real
+from typing import Any, Union
 
 from plasmapy.particles.exceptions import InvalidParticleError
 from plasmapy.particles.particle_class import CustomParticle, Particle
 from plasmapy.particles.particle_collections import ParticleList
 
 
+def _generate_error_message(args: tuple, kwargs: dict[str, Any]) -> str:
+    """Compose an error message for invalid particles."""
+
+    errmsg = "Unable to create a particle from: "
+
+    if args:
+        errmsg += repr(args)
+        if kwargs:
+            errmsg += " and "
+
+    if kwargs:
+        errmsg += repr(kwargs)
+
+    errmsg += (
+        ". For information on creating particles, see:"
+        "https://docs.plasmapy.org/en/stable/api/plasmapy.particles.ParticleLike.html"
```

Comment:
I'm a bit nervous about having a link in here that won't get updated if the namespace for `ParticleLike` changes, but I added a comment next to where `ParticleLike` is defined to look for this.